soundbay/models.py

```diff
@@ -29,6 +29,18 @@ def _choose_block_class(block):
         module_name = '.'.join(block.split('.')[:-1])
         return getattr(importlib.import_module(module_name), class_name)
 
+    def freeze_layers(self):
+        """
+        Freeze all layers except the classifier and last layer block from training, as a condition for finetune
+        """
+
+        for param in self.parameters():
+            param.requires_grad = False
+        for param in self.fc.parameters():
+            param.requires_grad = True
+        for param in self.layer4.parameters():
```

Comment:
This should not happen, as the implementation above is specific to all ResNet1Channel based models (those should all have the above attributes). I think if a future implementation of a new model would require to remove a layer block (like layer4), or change the classifier (FC) / add additional blocks, etc - the child model should have the method overloaded and designed for its own architecture

the method for models not based on ResNet1Channel is not yet implemented (it would require testing before implementation, to hopefuly find optimal finetune setting)